trollflow2/launcher.py

```diff
@@ -85,7 +85,7 @@ def get_test_message(test_message_file):
     return msg
 
 
-def check_results(produced_files, start_time, exitcode):
+def check_results(produced_files, start_time, exitcode, remote_filesystem_uri=None):
```

Comment:
This starts to get a bit complicated here with an extra parameter that refers to the produced files. Also, later in this module, the `_check_file` function is created, that contains if-statements to check all the cases (luckily just local and s3 for now, but that can grow).
To prepare for the future and make things clearer already now, I think we should introduce a new class interface, eg `SavedFile` for files to be saved that have a method for doing the check. In the current case we would have a subclass for local files and another for s3 files. That would allow to keep s3 stuff together in their own class and module, and polymorphism will be used to trigger the correct behaviour.

Reply:
Which class? The check isn't in any class at the moment.

Reply:
The kwarg is now removed in https://github.com/pytroll/trollflow2/pull/158/commits/42f6639d6887db0df88900c61a1aa3ac932295b6 . The filenames stored in `fmt` now have the eventual file location (as a side-effect of https://github.com/pytroll/trollflow2/pull/158/commits/1e8f88da4268f16c22e8475366e00300a1c5847b ) and `check_results()` can use those URIs directly. So the functionality stayed the same, but the interface is much cleaner.